packages/core/src/external/surescripts/fhir/bundle.ts

```diff
@@ -0,0 +1,45 @@
+import { Bundle, Medication } from "@medplum/fhirtypes";
+import { buildBundle } from "../../fhir/bundle/bundle";
+import { ResponseDetail } from "../schema/response";
+import { IncomingData } from "../schema/shared";
+import { initializeContext } from "./shared";
+import { getAllBundleEntries } from "./bundle-entry";
+
+import { dangerouslyDeduplicateFhir } from "../../../fhir-deduplication/deduplicate-fhir";
+import { hydrateFhir } from "../../fhir/hydration/hydrate-fhir";
+import { crosswalkNdcToRxNorm } from "../../term-server";
+import { NDC_URL } from "../../../util/constants";
+
+export async function convertIncomingDataToFhirBundle(
+  cxId: string,
+  patientId: string,
+  details: IncomingData<ResponseDetail>[]
+): Promise<Bundle> {
+  const bundle = buildBundle({ type: "collection", entries: [] });
+  const context = initializeContext(patientId);
+  for (const detail of details) {
+    const entries = getAllBundleEntries(context, detail);
+    bundle.entry?.push(...entries);
+  }
+  dangerouslyDeduplicateFhir(bundle, cxId, patientId);
+  await dangerouslyHydrateMedications(bundle);
+  await hydrateFhir(bundle, console.log);
```

Comment:
Def's not blocking this release, but it'd be good to have metrics on the components of the SS run at some point, in case it takes long we can tell what's the culprit

Reply:
Yes for sure! Added this onto my todo list for the upcoming PR with the scheduler.